hedera-node/hedera-app/src/main/java/com/hedera/node/app/authorization/PrivilegesVerifier.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.authorization;
+
+import static com.hedera.node.app.authorization.Authorizer.SystemPrivilege.AUTHORIZED;
+import static com.hedera.node.app.authorization.Authorizer.SystemPrivilege.IMPERMISSIBLE;
+import static com.hedera.node.app.authorization.Authorizer.SystemPrivilege.UNAUTHORIZED;
+import static com.hedera.node.app.authorization.Authorizer.SystemPrivilege.UNNECESSARY;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.file.SystemDeleteTransactionBody;
+import com.hedera.hapi.node.file.SystemUndeleteTransactionBody;
+import com.hedera.hapi.node.token.CryptoUpdateTransactionBody;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.authorization.Authorizer.SystemPrivilege;
+import com.hedera.node.config.ConfigProvider;
+import com.hedera.node.config.data.AccountsConfig;
+import com.hedera.node.config.data.FilesConfig;
+import com.hedera.node.config.data.LedgerConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+
+/**
+ * Checks whether an account is authorized to perform a system transaction that requires
+ * privileged access.
+ *
+ * <p>The checks in this class do not require access to state, and thus can be performed at any time.
+ */
+public class PrivilegesVerifier {
+
+    private final AccountsConfig accountsConfig;
+    private final FilesConfig filesConfig;
+    private final long numReservedSystemEntities;
+
+    @Inject
+    public PrivilegesVerifier(@NonNull final ConfigProvider configProvider) {
+        requireNonNull(configProvider, "configProvider cannot be null");
+
+        final var configuration = configProvider.getConfiguration();
+        this.accountsConfig = configuration.getConfigData(AccountsConfig.class);
+        this.filesConfig = configuration.getConfigData(FilesConfig.class);
+        this.numReservedSystemEntities =
+                configuration.getConfigData(LedgerConfig.class).numReservedSystemEntities();
+    }
+
+    /**
+     * Checks whether an account is exempt from paying fees.
```

Comment:
This description is incorrect. The method checks whether an account is authorized to perform this action.